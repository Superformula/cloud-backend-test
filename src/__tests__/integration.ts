import { createTestClient } from 'apollo-server-testing';
import gql from 'graphql-tag';

import { constructTestServer, mockedContext } from './__utils';


const GET_USER_BY_ID = gql`
query{
  users(
  	id: "fa8d7da0-af85-11eb-9859-577507baa6d7"
  ){
    items{
      id,
      name
    },
    lastEvaluatedKey
  }
}
`;



const mockedReadByIdDbResponse = {
  Items: [
    {
      name: 'New name',
      description: 'new desc',
      id: 'fa8d7da0-af85-11eb-9859-577507baa6d7',
      imageUrl: 'test.com'
    }
  ],
  Count: 1,
  ScannedCount: 21
};

describe('Queries', () => {
  it('user - read one by Id - success', async () => {
    
    const {server, geoDataStorage, storeDataStorage} = constructTestServer();

    mockedContext.db.scan.mockReturnValueOnce({
      promise: () => Promise.resolve(mockedReadByIdDbResponse)
    });

    const { query } = createTestClient(server);
    const res = await query({query: GET_USER_BY_ID});
    expect(res).toMatchSnapshot();
  });

  // it('fetches single launch', async () => {
  //   const {server, launchAPI, userAPI} = constructTestServer({
  //     context: () => ({user: {id: 1, email: 'a@a.a'}}),
  //   });

  //   launchAPI.get = jest.fn(() => [mockLaunchResponse]);
  //   userAPI.store = mockStore;
  //   userAPI.store.trips.findAll.mockReturnValueOnce([
  //     {dataValues: {launchId: 1}},
  //   ]);

  //   const {query} = createTestClient(server);
  //   const res = await query({query: GET_LAUNCH, variables: {id: 1}});
  //   expect(res).toMatchSnapshot();
  // });
});

// describe('Mutations', () => {
//   it('returns login token', async () => {
//     const {server, launchAPI, userAPI} = constructTestServer({
//       context: () => {},
//     });

//     userAPI.store = mockStore;
//     userAPI.store.users.findOrCreate.mockReturnValueOnce([
//       {id: 1, email: 'a@a.a'},
//     ]);

//     const {mutate} = createTestClient(server);
//     const res = await mutate({
//       mutation: LOGIN,
//       variables: {email: 'a@a.a'},
//     });
//     expect(res.data.login.token).toEqual('YUBhLmE=');
//   });

//   it('books trips', async () => {
//     const {server, launchAPI, userAPI} = constructTestServer({
//       context: () => ({user: {id: 1, email: 'a@a.a'}}),
//     });

//     // mock the underlying fetches
//     launchAPI.get = jest.fn();

//     // look up the launches from the launch API
//     launchAPI.get
//       .mockReturnValueOnce([mockLaunchResponse])
//       .mockReturnValueOnce([{...mockLaunchResponse, flight_number: 2}]);

//     // book the trip in the store
//     userAPI.store = mockStore;
//     userAPI.store.trips.findOrCreate
//       .mockReturnValueOnce([{get: () => ({launchId: 1})}])
//       .mockReturnValueOnce([{get: () => ({launchId: 2})}]);

//     // check if user is booked
//     userAPI.store.trips.findAll.mockReturnValue([{}]);

//     const {mutate} = createTestClient(server);
//     const res = await mutate({
//       mutation: BOOK_TRIPS,
//       variables: {launchIds: ['1', '2']},
//     });
//     expect(res).toMatchSnapshot();
//   });
// });
