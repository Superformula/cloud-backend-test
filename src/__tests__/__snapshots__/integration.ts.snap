// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries user - read one by Id - success 1`] = `
Object {
  "data": Object {
    "users": Object {
      "items": Array [
        Object {
          "id": "fa8d7da0-af85-11eb-9859-577507baa6d7",
          "name": "New name",
        },
      ],
      "lastEvaluatedKey": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
